test/Macros/SwiftifyImport/SizedBy/TypedPointer.swift

```diff
@@ -0,0 +1,108 @@
+                 .sizedBy(pointer: .return, size: "size"),
+                 .lifetimeDependence(dependsOn: .param(2), pointer: .return, type: .copy))
+func constReturnDependence(_ size: CInt, _ ptr: UnsafePointer<UInt8>) -> UnsafePointer<CChar> {}
+
+@_SwiftifyImport(.sizedBy(pointer: .param(2), size: "size"),
+                 .sizedBy(pointer: .return, size: "size"),
+                 .lifetimeDependence(dependsOn: .param(2), pointer: .return, type: .copy))
+func mutReturnDependence(_ size: CInt, _ ptr: UnsafeMutablePointer<UInt8>) -> UnsafeMutablePointer<UInt8> {}
+
+// CHECK:      @_alwaysEmitIntoClient @_disfavoredOverload
+// CHECK-NEXT: func constParam(_ ptr: UnsafeRawBufferPointer) {
+// CHECK-NEXT:     let size = CInt(exactly: unsafe ptr.count)!
+// CHECK-NEXT:     return unsafe constParam(ptr.baseAddress!.assumingMemoryBound(to: CChar.self), size)
+// CHECK-NEXT: }
+
+// CHECK:      @_alwaysEmitIntoClient @_disfavoredOverload
+// CHECK-NEXT: func mutParam(_ ptr: UnsafeMutableRawBufferPointer) {
+// CHECK-NEXT:     let size = CInt(exactly: unsafe ptr.count)!
+// CHECK-NEXT:     return unsafe mutParam(ptr.baseAddress!.assumingMemoryBound(to: UInt8.self), size)
+// CHECK-NEXT: }
+
+// CHECK:      @_alwaysEmitIntoClient @_disfavoredOverload
+// CHECK-NEXT: func exprParam(_ ptr: UnsafeMutableRawBufferPointer, _ size: CInt, _ count: CInt) {
+// CHECK-NEXT:     let _ptrCount = unsafe ptr.count
+// CHECK-NEXT:     if _ptrCount != size * count {
+// CHECK-NEXT:       fatalError("bounds check failure in exprParam: expected \(size * count) but got \(_ptrCount)")
+// CHECK-NEXT:     }
+// CHECK-NEXT:     return unsafe exprParam(ptr.baseAddress!.assumingMemoryBound(to: UInt8.self), size, count)
+// CHECK-NEXT: }
+
+// CHECK:      @_alwaysEmitIntoClient @_disfavoredOverload
+// CHECK-NEXT: func constReturn(_ size: CInt) -> UnsafeRawBufferPointer {
+// CHECK-NEXT:     return unsafe UnsafeRawBufferPointer(start: unsafe constReturn(size), count: Int(size))
+// CHECK-NEXT: }
+
+// CHECK:      @_alwaysEmitIntoClient @_disfavoredOverload
+// CHECK-NEXT: func mutReturn(_ size: CInt) -> UnsafeMutableRawBufferPointer {
+// CHECK-NEXT:     return unsafe UnsafeMutableRawBufferPointer(start: unsafe mutReturn(size), count: Int(size))
+// CHECK-NEXT: }
+
+// CHECK:      @_alwaysEmitIntoClient @_disfavoredOverload
+// CHECK-NEXT: func exprReturn(_ size: CInt, _ count: CInt) -> UnsafeMutableRawBufferPointer {
+// CHECK-NEXT:     return unsafe UnsafeMutableRawBufferPointer(start: unsafe exprReturn(size, count), count: Int(size * count))
+// CHECK-NEXT: }
+
+// CHECK:      @_alwaysEmitIntoClient @_disfavoredOverload
+// CHECK-NEXT: func constParamNoreturn(_ ptr: RawSpan) {
+// CHECK-NEXT:     let size = CInt(exactly: ptr.byteCount)!
+// CHECK-NEXT:     return unsafe ptr.withUnsafeBytes { _ptrPtr in
+// CHECK-NEXT:       return unsafe constParamNoreturn(_ptrPtr.baseAddress!.assumingMemoryBound(to: CChar.self), size)
```

Comment:
@atrick The exact memory model around rebinding memory is a bit unclear to me. Is this a safe way to cast a the base pointer of a `RawSpan` to a `const char * __sized_by(...)`?

This is _only_ done when the size of the pointee type is 1 byte on the C side fwiw, so the assumption is that the C API will access memory through a byte-sized pointer, which should be fine regardless of what size it was written as without violating strict aliasing in C's memory model. When the pointee size is greater than 1 byte we ignore `__sized_by`, so `__counted_by` would have to be used, which won't result in a RawSpan on the Swift side.

Reply:
That looks good to me. It's always safe to *assume* an UnafeRawPointer is `UnsafePointer<CChar>` if the resulting pointer is used in C/C++ code (`char*` is like a raw pointer in C). In fact, if you were directly calling C, you wouldn't need `assumingMemoryBound(to:)`:

`constParamNoreturn(_ptrPtr.baseAddress!, _ptrPtr.count)`

If you wanted to view the raw pointer as `UnsafeBufferPointer<CChar>` in some arbitrary Swift code, where other Swift code may be accessing the same memory as a different type, then you could create an optimization barrier around that temporary scope using:
`_ptrPtr.withMemoryRebound(to: CChar.self) {...}`

Reply:
Aha, thank you, that clarifies things! I think I'll keep the `assumingMemoryBound(to:)` around since it makes it easier to test by attaching the macro to a Swift function.